### PR TITLE
HackStudio: Add a new commit dialog and use String instead of LexicalPath

### PR DESCRIPTION
**Pull request by cbyrneee**
### HackStudio: Add new multiline commit dialog
The previous commit dialog was quite lacking in terms of input, with a single input box without multiline editing support. This commit improves the dialog by adding a multiline text editor and a line and column counter.

*This commit dialog is mostly the same as the one which was featured in one of my previous PRs a few months ago!*

### HackStudio: Use ``String`` instead of ``LexicalPath``
``LexicalPath`` is a 'heavier' object than a ``String`` that is mainly used for  path parsing and validation, we don't actually need any of that in ``GitRepo`` and its related files, so let's move to ``String``.

### CODE REVIEW

File: Userland/DevTools/HackStudio/Dialogs/Git/GitCommitDialog.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#include <LibGUI/Dialog.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/TextEditor.h>
+#include <LibGUI/Window.h>
+
+namespace HackStudio {
+
+using OnCommitCallback = Function<void(StringView const& message)>;
+
+class GitCommitDialog : public GUI::Dialog {
```
Comment (BertalanD) [inline Userland/DevTools/HackStudio/Dialogs/Git/GitCommitDialog.h:20]: Can this class be made `final`?

**Review by BertalanD — CHANGES_REQUESTED** [review]
Great work!

Just some small nitpicks :^)

File: Userland/DevTools/HackStudio/Dialogs/Git/GitCommitDialog.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#include <AK/RefPtr.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/Dialog.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/TextEditor.h>
+#include <LibGUI/Window.h>
+
+namespace HackStudio {
+
+using OnCommitCallback = Function<void(StringView const& message)>;
```
[the remark this answers is not in the document]
Comment (cbyrneee) [reply Userland/DevTools/HackStudio/Dialogs/Git/GitCommitDialog.h]: Done + leaving a note from Discord here
image

**Review by BertalanD — APPROVED** [review]
LGTM!

**Pull request merged**